auth/authn-authz-intro.md

```diff
@@ -0,0 +1,37 @@
+
+The OpenConfig project initially provided means to inter-work with existing means of authorization and authentication on network devices - providing means to supply a username and password via gRPC metadata for gNMI and gNOI - along with mutual TLS authentication. However, this approach has a number of limitations:
+
+* It relies on an external run-time dependency to ensure that authentication/authorization is performed by the switch - particularly, the external AAA server being accessible. Thus introducing new failure modes that need to handled - such as fall-back credentials.
+* It introduces a potential vulnerability if the remote AAA is subject to MITM attacks.
+* In high-volume deployments (hundreds of thousands of switches), it introduces a new service that must be scaled.
+* It can introduce external factors which impact the performance of programmatic RPCs to the device - e.g., the latency between the switch and the AAA server.
+* It requires extension and definition of new patterns to be able to handle more granular authorization of access to the configuration and telemetry stored on a network device.
+* It requires a networking-specific approach to be taken for secure access to device endpoints rather than building on the wider industry open source developments, and their surrounding ecosystem.
+
+## Proposed Direction
+
+We propose to develop an approach which addresses a number of these concerns, particularly to provide:
+
+* A secure means by which authentication and authorization policy can be provided to a network device and stored locally.
+* Means by which this policy can interact with user identifiers that are specified as part of the gRPC connection - ideally building upon wider industry developments.
+* Means by which this policy can be used as a mechanism for authenticating users against other access mechanisms (e.g., local console, SSH) such that an operator can unify their authentication and authorization policy for network devices into a single definition.
+* Means within the policy to provide for granular access to a data tree which is described by a specified data model. This will particularly focus around the YANG tree that is defined by the OpenConfig data models.
+
+We envisage that this approach can coexist with authentication and authorization mechanisms that are already implemented by typical NOS today, whilst also evolving towards a unified, modernised approach.
```

Comment:
By "coexist" I think we need to be more specific that if the new APIs are supported and in effect, control of the overlapped configurations via gNMI are required to be disabled.

Reply:
Thanks - I added a statement into the doc to say this. PTAL.